### PR TITLE
Refactor event-patterns, use MediaSessions API

### PR DESCRIPTION
**Pull request by rtshkmr**
This PR does a bunch of things: 
1. refactor of events bridge 
2. use media sessions api

## Migration Notes:
- most recent db state dump (for migration):
[1723306013568018000.json](https://github.com/user-attachments/files/16570171/1723306013568018000.json)

### CODE REVIEW

**Comment by rtshkmr** [comment]
Here's a migrator json dump that can be used. Contains the two videos: 

[1722313237078948000.json](https://github.com/user-attachments/files/16422047/1722313237078948000.json)


will generate another one once the content of the metadata is clear

**Comment by rtshkmr** [comment]
[1722398151031913000.json](https://github.com/user-attachments/files/16436060/1722398151031913000.json)

Contains some metadata about the two voice files we have

File: assets/js/hooks/audio_player.js [hunk 1]
```diff
@@ -191,6 +188,46 @@ AudioPlayer = {
       this.player.play() // force a play event if is not paused
     }
   },
+  updateMediaSession(playback) {
+    const isSupported = "mediaSession" in navigator;
```
Comment (ks0m1c) [inline assets/js/hooks/audio_player.js:192]:
```
  // title, artist, album, artwork, ...
  chapterInfo: [{
    title: 'Verse 1',
    startTime: 0,
    artwork: [
      { src: 'https://via.placeholder.com/128', sizes: '128x128', type: 'image/png' },
      { src: 'https://via.placeholder.com/512', sizes: '512x512', type: 'image/png' },
    ]
  }, {
    title: 'Verse 2',
    startTime: 42,
    artwork: [
      { src: 'https://via.placeholder.com/128', sizes: '128x128', type: 'image/png' },
      { src: 'https://via.placeholder.com/512', sizes: '512x512', type: 'image/png' },
    ]
  }]
});
```
chapter info handler

**Comment by ks0m1c** [comment]
```
9 4.864 12:26:30.115 [debug] Downloading esbuild from https://registry.npmjs.org/@esbuild/linux-x64/0.17.11
#19 5.254 ✘ [ERROR] Expected "}" but found "createMediaMetadata"
#19 5.254 
#19 5.254     js/hooks/audio_player.js:199:2:
#19 5.254       199 │   createMediaMetadata(playback) {
#19 5.254           │   ~~~~~~~~~~~~~~~~~~~
#19 5.254           ╵   }
#19 5.254 
#19 5.262 1 error
#19 5.272 ** (Mix) `mix esbuild default --minify --loader:.ttf=file` exited with 1
#19 ERROR: process "/bin/sh -c mix assets.deploy" did not complete successfully: exit code: 1
------
 > [builder 13/17] RUN mix assets.deploy:
4.864 12:26:30.115 [debug] Downloading esbuild from https://registry.npmjs.org/@esbuild/linux-x64/0.17.11
Error:  [ERROR] Expected "}" but found "createMediaMetadata"
5.254 
5.254     js/hooks/audio_player.js:199:2:
5.254       199 │   createMediaMetadata(playback) {
5.254           │   ~~~~~~~~~~~~~~~~~~~
5.254           ╵   }
5.254 
5.262 1 error
```

File: assets/js/hooks/floater.js [hunk 2]
```diff
@@ -2,6 +2,12 @@
  * Ideally generic hook for floating logic.
  */
 import {isMobileDevice} from "../utils/uncategorised_utils.js";
+import {
```
Comment (ks0m1c) [inline assets/js/hooks/floater.js:5]: post vendor floater import g

File: lib/vyasa_web/components/audio_player.ex [hunk 3]
```diff
@@ -1,38 +1,43 @@
...
-      <div id="audio-player" phx-hook="AudioPlayer">
-        <audio></audio>
-      </div>
-      """
-    end
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div id="audio-player" phx-hook="AudioPlayer">
+      <audio data-playback={Jason.encode!(@playback)}></audio>
```
Comment (ks0m1c) [inline lib/vyasa_web/components/audio_player.ex:13]: ah the source of the data-playback change

**Review by ks0m1c — COMMENTED** [review]
hmm not much changes here just a lot of linting looks g

File: lib/vyasa/medium.ex [hunk 4]
```diff
@@ -137,17 +136,18 @@ defmodule Vyasa.Medium do
   def get_event!(id), do: Repo.get!(Event, id)
 
   def get_event_by_order!(%Event{origin: origin, voice_id: v_id}, order) do
-    #TODO merge Sangh Filters to fix -1 order case for origin backwards operators
-    (from e in Event,
+    # TODO merge Sangh Filters to fix -1 order case for origin backwards operators
```
Comment (rtshkmr) [inline lib/vyasa/medium.ex:139]: @ks0m1c reminder on this TODO

Comment (rtshkmr) [reply lib/vyasa/medium.ex:139]: is this still relevant?

Comment (ks0m1c) [reply lib/vyasa/medium.ex:139]: this was actually built out for the admin view idea before Repo.Paginated.all(opts) was merged. with Vyasa.Repo.Paginated module filter is expressed as

[filter: String.to_integer(cursor) |> DateTime.from_unix!(:millisecond), sort_attribute: :origin]

**Pull request merged**